src/Chapter2.hs

```diff
@@ -842,7 +866,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate a l
+  | a < 0 = []
+  | a == 0 = l
+  | l == [] = []
+  | otherwise = take (length l) (drop a (cycle l))
```

Comment:
Amazing 👍🏼 

If you want to be sneaky, you can also drop not just `a` but `a` mod by the length of the list so it is a bit more efficient in case when `a` is bigger than the length👌🏼